doc-gen/Makefile

```diff
@@ -0,0 +1,37 @@
+
+DEVICES = \
+    pca10040 \
+    microbit \
+    nrf52840-mdk \
+    pca10031 \
+    bluepill \
+    arduino \
+    digispark \
+    reelboard \
+    pca10056 \
+    arduino-nano33 \
+    itsybitsy-m0 \
+    feather-m0 \
+    feather-m4 \
+    trinket-m0 \
+    metro-m4-airlift \
+    circuitplay-express \
+    stm32f4disco \
+    circuitplay-express \
+    gameboy-advance \
+    itsybitsy-m4 \
+    nucleo-f103rb \
+    hifive1b \
+    pinetime-devkit0 \
+    x9pro
```

Comment:
I see you reordered some of these devices, any reason for that?
The order doesn't really matter, but I kept it at the same order as the smoke tests in the Makefile, for easier updating (maybe?).

Reply:
I was going to put it all into alpha order, but instead I just added the missing ones in semi-random order.